lazyflow/utility/io_util/OMEZarrStore.py

```diff
@@ -256,13 +257,42 @@ def _get_zarr_cache_max_size() -> int:
     return math.floor(caches_max * permissible_fraction_max)
 
 
-def _unescape_and_get_store(uri: str, mode="r", **kwargs) -> FSStore:
+def _ensure_connection_and_get_store(uri: str, mode="r", **kwargs) -> FSStore:
     if uri.startswith("file:"):
         # Zarr's FSStore implementation doesn't unescape file URLs before piping them to
         # the file system. We do it here the same way as in pathHelpers.uri_to_Path.
         # Primarily this is to deal with spaces in Windows paths (encoded as %20).
         uri = os.fsdecode(unquote_to_bytes(uri))
-    return FSStore(uri, mode=mode, **kwargs)
+    store = FSStore(uri, mode=mode, **kwargs)
+    try:
+        store["test"]
+        # Any error not handled here is either a successful connection (404 for "test"), or an unknown problem
```

Comment:
not sure I understand the motivation here. So you try to access the key "test" in the store to test if you have access to the store in general? What if the object behind "test" is a 10GB file? "test" is probably not the most uncommon random key for stuff (I at least at some point had large random files called "test" around to ... test stuff).

Reply:
Yeah should go for a key with higher chance to go 404...